src/api/services/auditTrailService.ts

```diff
@@ -0,0 +1,35 @@
+import { AuditTrail, AuditTrailEvents, SharingAction } from '../entities/AuditTrail';
+import { Participant } from '../entities/Participant';
+import { getLoggers } from '../helpers/loggingHelpers';
+
+export const insertSharingAuditTrails = async (
+  participant: Participant,
+  userId: number,
+  userEmail: string,
+  action: SharingAction,
+  siteIds: number[]
+) => {
+  try {
+    const sharingAuditTrail = {
+      participantId: participant.id,
+      userId,
+      userEmail,
+      event: AuditTrailEvents.UpdateSharingPermissions,
+      eventData: {
+        siteId: participant.siteId,
+        action,
+        sharingPermissions: siteIds,
+      },
+    } as AuditTrail;
```

Comment:
Try to avoid `as` whenever possible - in this case, it's masking a possible bug (you can pass in a participant which doesn't have a siteId). When you use `as` you're telling TypeScript to skip checking the types (although sometimes the linter still finds things).

Better to declare the variable with a type - e.g. `const sharingAuditTrail: AuditTrail = {`. That way TypeScript knows you expect something of type AuditTrail and it'll check that what you assign matches.